anker/src/state.rs

```diff
@@ -284,26 +356,10 @@ impl Anker {
         anker_program_id: &Pubkey,
         accounts: &SellRewardsAccountsInfo,
     ) -> ProgramResult {
+        // Check if the token swap account is the same one as the stored in the instance.
+        self.check_token_swap_pool(accounts.token_swap_pool)?;
+        let token_swap = get_token_swap_instance(accounts.token_swap_pool)?;
```

Comment:
That would prevent us from using the `get_token_swap_instance` to get, for instance, the `new_token_swap` when changing the instance.

Reply:
Hmm yeah. We could still do it, but then we need to shulffle some arguments around. Deserialize the old one first, then overwrite `self.token_swap_pool`, then do the check, and only then save. Then the check can take the old and new instance as arguments. But then you have this code where the check comes after the mutation, which works because it runs in a transaction, but it can still be a bit counter-intuitive. I’ll leave it up to you then.